File: Search Algorithms/linearSearch.cpp
#include <iostream>
using namespace std;

// Runtime: O(n)
int linearSearch(int arr[], int n, int x) {
    int i = 0;
    for (i; i < n; i++)
    {
        if (arr[i] == x) {
            return i;
        }
    }
    return - 1;
}

// driver 
int main(void) {
    int arr[] = {2, 3, 4, 10, 40};
    int x = 10;
    int n = sizeof(arr) / sizeof(arr[0]);

    int result = linearSearch(arr, n, x);
    (result == -1)
        ? cout << "Element is not present in array"
        : cout << "Element is present at index " << result;
    return 0;
}

File: Search Algorithms/linear_search.py
# Linear Search

# Runtime: O(n)
def linear_search(arr, n, x):
    for i in range(0, n):
        if arr[i] == x:
            return i
    return - 1

# driver code to run tests
def main():
    arr = [10, 20, 80, 30, 60, 50, 110, 100, 130, 170]
    vals = [110, 175, 10, 15]

    for x in vals:
        n = len(arr)
        result = linear_search(arr, n, x)
        if result > 0:
            print(result)
            print(f"{x} is present at index {result}")
            print()
        else:
            print(result)
            print(f"{x} is not present in arr")
            print()

if __name__ == '__main__':
    main()


    